src/cli/rustup_mode.rs

```diff
@@ -911,31 +925,20 @@ fn check_updates(cfg: &Cfg) -> Result<utils::ExitCode> {
         }
     }
 
-    // Get current rustup version
-    let current_version = env!("CARGO_PKG_VERSION");
-
-    // Get available rustup version
-    let available_version = get_available_rustup_version()?;
-
-    let _ = t.attr(term2::Attr::Bold);
-    write!(t, "rustup - ")?;
+    check_rustup_update()?;
 
-    if current_version != available_version {
-        let _ = t.fg(term2::color::YELLOW);
-        write!(t, "Update available")?;
-        let _ = t.reset();
-        writeln!(t, " : {} -> {}", current_version, available_version)?;
-    } else {
-        let _ = t.fg(term2::color::GREEN);
-        write!(t, "Up to date")?;
-        let _ = t.reset();
-        writeln!(t, " : {}", current_version)?;
-    }
     Ok(utils::ExitCode(0))
 }
 
 fn update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
-    let self_update = !m.is_present("no-self-update") && !self_update::NEVER_SELF_UPDATE;
+    let self_update_mode = cfg.get_self_update_mode()?;
+    // Priority: no-self-update feature > self_update_mode > no-self-update args.
```

Comment:
This should be documented somewhere users can see it; since users need to reason about it.

Reply:
The documentation now mentions them in order of priority.